packages/agent/src/prototyping/clients/dwn-server-info-cache-memory.ts

```diff
@@ -0,0 +1,79 @@
+
+import ms from 'ms';
+import { TtlCache } from '@web5/common';
+import { DwnServerInfoCache, ServerInfo } from './server-info-types.js';
+
+/**
+ * Configuration parameters for creating an in-memory cache for DWN ServerInfo entries.
+ *
+ * Allows customization of the cache time-to-live (TTL) setting.
+ */
+export type DwnServerInfoCacheMemoryParams = {
+  /**
+   * Optional. The time-to-live for cache entries, expressed as a string (e.g., '1h', '15m').
+   * Determines how long a cache entry should remain valid before being considered expired.
+   *
+   * Defaults to '15m' if not specified.
+   */
+  ttl?: string;
+}
+
+export class DwnServerInfoCacheMemory implements DwnServerInfoCache {
```

Comment:
Consider turning this into generic, seems low hanging fruit.

```suggestion
export class InMemoryCache<T> implements GenericCache<T> {
```

Reply:
The thing about this is we already have one called MemoryStore which is why his is more specific. Maybe MemoryStoreAsync? Unsure

Reply:
@thehenrytsai Yeah @shamilovtim  called out something similar and I was considering this, but figured we could do a separate PR to add `TTLCacheAsync` or something of that sort to `@web5/common` because we are doing this same wrapper in a couple of places to conform with an async interface.